src/libaktualizr/storage/invstorage.h

```diff
@@ -184,6 +184,7 @@ class INvStorage {
   virtual void saveEcuReportCounter(const Uptane::EcuSerial& ecu_serial, int64_t counter) = 0;
   virtual bool loadEcuReportCounter(std::vector<std::pair<Uptane::EcuSerial, int64_t>>* results) = 0;
 
+  virtual bool checkAvailableDiskSpace(uint64_t required_bytes) const = 0;
```

Comment:
one minor remark, regarding the "checkAvailableDiskSpace" function, is it really necessary to make this function pure virtual in the "invstorage" interface? 

I suspect this function will be called quite often, does it make sense to increase virtual table?

Reply:
I'm not sure I have a choice here. We only have one fully functional storage implementation at present, but I think this depends on the specific implementation. For SQL, we store the files on disk in the same directory as the database, so I know I can check the filesystem that all that lives on. But a different implementation might do something else, like use multiple partitions, and then my current logic would have to be adapted to that.